ports/servoshell/desktop/app.rs

```diff
@@ -329,26 +329,79 @@ impl App {
+
+                    if let Err(error) = sender.send(context.is_some()) {
+                        warn!("Failed to send response of IsWebViewOpein: {error}");
+                    }
                 },
                 webdriver_msg @ WebDriverCommandMsg::IsBrowsingContextOpen(..) => {
                     running_state.forward_webdriver_command(webdriver_msg);
                 },
+                WebDriverCommandMsg::NewWebView(response_sender, load_status_sender) => {
+                    let new_webview =
+                        running_state.create_toplevel_webview(Url::parse("auto:blank").unwrap());
+
+                    if let Err(error) = response_sender.send(new_webview.id()) {
+                        warn!("Failed to send response of NewWebview: {error}");
+                    }
+
+                    running_state.set_load_status_sender(new_webview.id(), load_status_sender);
+                },
                 WebDriverCommandMsg::CloseWebView(webview_id) => {
                     running_state.close_webview(webview_id);
                 },
-                WebDriverCommandMsg::GetWindowSize(..) |
-                WebDriverCommandMsg::FocusWebView(..) |
+                WebDriverCommandMsg::FocusWebView(webview_id) => {
+                    if let Some(webview) = running_state.webview_by_id(webview_id) {
+                        webview.focus();
+                    }
+
+                    // TODO: send a response to the WebDriver
+                    // so it knows when the focus has finished.
+                },
+                WebDriverCommandMsg::GetWindowSize(_webview_id, response_sender) => {
+                    let window = self
+                        .windows
+                        .values()
+                        .next()
+                        .expect("Should have at least one window in servoshell");
+
+                    if let Err(error) = response_sender.send(window.screen_geometry().size) {
+                        warn!("Failed to send response of GetWindowSize: {error}");
+                    }
+                },
+                WebDriverCommandMsg::SetWindowSize(webview_id, size, size_sender) => {
+                    let Some(webview) = running_state.webview_by_id(webview_id) else {
+                        continue;
+                    };
+
+                    let window = self
+                        .windows
+                        .values()
+                        .next()
+                        .expect("Should have at least one window in servoshell");
+
+                    let size = window.request_resize(&webview, size);
+                    if let Err(error) = size_sender.send(size.unwrap_or_default()) {
+                        warn!("Failed to send window size: {error}");
+                    }
+                },
+                webdriver_msg @ WebDriverCommandMsg::GetViewportSize(..) => {
+                    running_state.forward_webdriver_command(webdriver_msg);
```

Comment:
You should be able to retrieve the size of the viewport here without forwarding the message. The viewport of a `WebView` is `window.rendering_context().size2d()`. I think it makes sense to implement this here without forwarding or just waiting to handle this in a followup.

Reply:
Okay. Sounds good. For now, just don't move this message to the embedder yet. I can review the change very quickly.